### PR TITLE
Lesson 3

### PR DESCRIPTION
**Pull request by Pizzabagels**
Merging lesson 3 mailroom homeworks

### CODE REVIEW

File: students/Tommy Aguilu/mailroom_part2.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            send_many(donor_raw)
+        else:
+            donation_list = []
+            donation = input("Donor not found in list please specify how donation amount")
+            donation_list.append(donation)
+            donor_raw.update({donor_choice : donation_list})
+            donor_letter(donor_raw,donor_choice)
+    elif choice == str(3):
+        print("closing out")
+        sentinal = False
```
Comment (majstorpoki) [inline students/Tommy Aguilu/mailroom_part2.py:65]: Intelligent and neat

File: students/Tommy Aguilu/mailroom_part2.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                f.close()
+    print("letters sent!")
+
+
+for key, value in donor_raw.items():
+    x = []
+    x.append(key)
+    x.append(sum(value)) #total donations
+    x.append(len(value)) #number of donations
+    x.append((sum(value)/len(value)).__round__()) #avg donations
```
Comment (majstorpoki) [inline students/Tommy Aguilu/mailroom_part2.py:42]: nice

File: students/Tommy Aguilu/mailroom_part2.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        for k,v  in donor_raw.items():
+            with open(k, "w") as f:
+                length = (len(v)-1)
+                f.write(("thank you {} for your most recent donation of {}".format(k, v[length])))
+                f.close()
+    print("letters sent!")
+
+
+for key, value in donor_raw.items():
+    x = []
```
Comment (majstorpoki) [inline students/Tommy Aguilu/mailroom_part2.py:38]: Nice, list of lists for processed donors

File: students/Tommy Aguilu/mailroom_part2.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
+### Author - Tommy Aguilu
```
Comment (majstorpoki) [inline students/Tommy Aguilu/mailroom_part2.py:1]: Update mailroom with file writing.
Goal: Write a full set of letters to all donors to individual files on disk.

Where is this accomplished?

**Review by majstorpoki — CHANGES_REQUESTED** [review]
Did you accomplish file write functionality in pt 2?

**Comment by Pizzabagels** [comment]
Hey Dr. Pokrajac,

Yes writing a full set of letters based on the donor list is in the
function send_many. In that function it writes a file named after the key
of the value in the list and then writes a string with the key and last
value in the list inserted into a predetermined sentence in to the file it
just created. This is found on lines 27-34.


    for k,v  in donor_raw.items():
        with open(k, "w") as f:
            length = (len(v)-1)
            f.write(("thank you {} for your most recent donation of
{}".format(k, v[length])))
            f.close()

**Review by majstorpoki — APPROVED** [review]

**Pull request merged**